lib/CommandParser.js

```diff
@@ -108,9 +110,40 @@ class CommandParser {
     throw new InvalidCommandError();
   }
 
+  /**
+   * Check command for partial match on primary directions, or exact match on secondary name or abbreviation
+   * @param {string} command
+   * @return {?string}
+   */
+  static checkMovement(command)
+  {
+    const primaryDirections = ['north', 'south', 'east', 'west', 'up', 'down'];
+
+    for (const direction of primaryDirections) {
+      if (direction.indexOf(command) === 0) {
+        return direction;
+      }
+    }
+
+    const secondaryDirections = [
+      { abbr: 'ne', name: 'northeast' },
+      { abbr: 'nw', name: 'northwest' },
+      { abbr: 'se', name: 'southeast' },
+      { abbr: 'sw', name: 'southwest' }
+    ];
+
+    for (const direction of secondaryDirections) {
+      if (direction.abbr === command || direction.name === command) {
```

Comment:
```suggestion
      if (direction.abbr === command || direction.name.indexOf(command) === 0) {
```

Would this not make that case work?